cloudbrain/core.go

```diff
@@ -140,6 +145,45 @@ func (c *Core) CreateInstance(ctx context.Context, providerName string, attr Cre
 	}, nil
 }
 
+// RemoveInstance creates an instance in the database and queues off the cloud
+// create job in the background.
+func (c *Core) RemoveInstance(ctx context.Context, attr DeleteInstanceAttributes) (*Instance, error) {
+	inst, err := c.db.GetInstance(attr.InstanceID)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = c.db.RemoveInstance(inst)
+	if err != nil {
+		cbcontext.LoggerFromContext(ctx).WithFields(logrus.Fields{
```

Comment:
This logger should also be removed, I think, and replace `return nil, err` with `return nil, errors.Wrap(err, "failed to delete instance from database")` or something like that.